flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogTable.java

```diff
@@ -46,4 +46,11 @@
 	 * @return a map of properties
 	 */
 	Map<String, String> toProperties();
+
+	/**
+	 * Returns a copy of this {@code CatalogTable} with given table options {@code options}.
+	 *
+	 * @return a new copy of this table with replaced table options
+	 */
+	CatalogTable copy(Map<String, String> options);
```

Comment:
use `properties` to be aligned with other places?

Reply:
As discussed in the mailing list, `options` is our new preference to describe the table properties.